pkg/asset/agent/image/oc.go

```diff
@@ -64,10 +74,29 @@ const (
 	templateImageExtractWithIcsp = "oc image extract --path %s:%s --confirm --icsp-file=%s %s"
 )
 
-// Get the CoreOS ISO from the releaseImage
-func (r *release) GetBaseIso(releaseImage, pullSecret, architecture string, mirrorConfig []mirror.RegistriesConfig) (string, error) {
+// ExtractFile extracts the specified file from the given image name, and store it in the cache dir.
+func (r *release) ExtractFile(image string, filename string, architecture string) (string, error) {
+	imagePullSpec, err := r.getImageFromRelease(image)
```

Comment:
I had trouble using the function, but when I changed image to machineOsImageName here, the function worked. Does that sound right to you?
```suggestion
	imagePullSpec, err := r.getImageFromRelease(machineOsImageName)
```

Reply:
Disregard this comment, as Andrea explained to me, the image parameter should be the name of an image as it is in the release image. The machineOsImageName is coincidentally one of the images in the payload.